src/patterns/OcFileActions.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <div uk-offcanvas="mode: slide" class="uk-offcanvas uk-offcanvas-bottom oc-file-actions">
+    <div class="uk-offcanvas-bar">
+      <span v-text="_label"></span>
+      <ul class="uk-nav">
+        <template v-for="(action, i) in actions">
+          <li uk-toggle="target:.oc-file-actions" :key="i" @click="action.onClick">
+            <a class="uk-inline">
+              <oc-icon v-if="action.icon" :name="action.icon"></oc-icon>
+              <span class="uk-text-top" v-text="action.label" />
+            </a>
+          </li>
+        </template>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<style>
+.uk-offcanvas.uk-offcanvas-bottom {
+  left: auto;
+  height: 100vh;
+}
+
+.uk-offcanvas.uk-offcanvas-bottom > .uk-offcanvas-bar {
+  width: 100vw;
+  transform: translateX(0) translateY(100%);
+  top: auto;
+}
+
+.uk-offcanvas.uk-offcanvas-bottom.uk-open > .uk-offcanvas-bar {
+  transform: translateX(0) translateY(0);
+}
+</style>
+
+<script>
+/**
+ * Create file actions.
+ */
+export default {
+  name: "oc-file-actions",
+  status: "prototype",
+  release: "1.0.0",
+  props: {
+    /**
+     * The label to display.
+     */
+    label: {
+      type: String,
+      required: false,
+      default: "Open %s in",
+    },
+    /**
+     * The filename to display.
+     */
+    filename: {
+      type: String,
+      required: false,
+      default: "",
+    },
+    /**
+     * The actions from which the user can select
+     */
+    actions: {
+      type: Array,
+      required: true,
+      default: function() {
+        return []
+      },
+    },
+  },
+  computed: {
+    _label() {
+      return this.label.replace(/\%s/, this.filename)
+    },
+  },
+}
+</script>
+
+<docs>
+```jsx
+  <template>
+    <div>
+      <oc-button uk-toggle="target: #file-actions" text="click to open actions"/>
```

Comment:
we need to get rid of the uk-toggle - see the oc-dialog component - THX

Reply:
oc-dialog makes the content of the dialog explicit in template. so it can set up the handlers properly. The fileactions are currently more dynamic, rendering the content based on the action objects. I can revert that and follow the approach the dialog takes, but then using it will no longer be as simple as
`oc-file-actions name="demoActions" :filename="filename" :actions="actions">`.
I mean it isn't anyway ... 

Furthermore, the uikit offcanvas can be closed by clicking outside of it. It fires a hide event, and we can use that to change the internal state accordingly. 

What am I doing wrong? I just went the file actions offcanvas to pop in when a button is clicked. Why do I need to specify anything that has to do with closing the offcanvas in the example at all? that should be self contained, shouldn't it?

Reply:
how this should work from my understanding:

```html
<button @click='showFileActions=true'/>
<oc-file-actions v-model='showFileActions'/>

Reply:
but we can address this later as well ... just squash commits and we merge this
this is fine for the first approach

Reply:
@DeepDiver1975 I went over this with @felixheidecke. We moved this to a global event system. There is a more beautiful solution for global events, but I consider that outside this PR.